main.py

```diff
@@ -40,55 +40,63 @@ def make_menu_decision():
-            if (len(password) < 8):
-                    errors.append("Password length must be above 8 characters.")
-            if not re.search("[a-z]", password):
-                errors.append("Password must contain at least one lowercase letter.")
-            if not re.search("[A-Z]", password):
-                errors.append("Password must contain at least one uppercase letter.")
-            if not re.search("[0-9]", password):
-                errors.append("Password must contain at least one number.")
-            if not re.search("[_@$]", password):
-                errors.append("Password must contain at least one special character.")
-            # This enables an error if any whitespace characters are detected. I don't think it's used correctly, though.
-            # String constants need an 'r' prefix, so I think it should be (r"\s", password)
-            if re.search("\s", password):
-                errors.append("No whitespace characters are allowed.")
+    while check_emp_file() is True:
+        display_password_instructions()
+        password = input('Enter new password: ')
+        errors = []  # Initialize this with zero elements. If an error occurs, then add it to array
+        if (len(password) < 8):
+                errors.append("Password length must be above 8 characters.")
+        if not re.search("[a-z]", password):
+            errors.append("Password must contain at least one lowercase letter.")
+        if not re.search("[A-Z]", password):
+            errors.append("Password must contain at least one uppercase letter.")
+        if not re.search("[0-9]", password):
+            errors.append("Password must contain at least one number.")
+        if not re.search("[_@$]", password):
+            errors.append("Password must contain at least one special character.")
+        # This enables an error if any whitespace characters are detected. I don't think it's used correctly, though.
+        # String constants need an 'r' prefix, so I think it should be (r"\s", password)
+        if re.search("\s", password):
+            errors.append("No whitespace characters are allowed.")
 
-            if (len(errors) > 0):  # If errors contains ANY elements, then print the errors and loop through again.
-                print("The following error(s) occured.")
-                # Print each error using a new line to separate each error
-                for error in errors:
-                    print(error)
-            else:
-                print("Valid password and stored")
-                break
-                
+        if (len(errors) > 0):  # If errors contains ANY elements, then print the errors and loop through again.
+            print("The following error(s) occured.")
+            # Print each error using a new line to separate each error
+            for error in errors:
+                print(error)
+        else:
+            with open('empPass.txt', 'a') as f:
```

Comment:
Thing to remember with IO (input/output) in regard to files is an exception can occur. 
What if the file is opened by another user, e.g open in notepad...?   This could throw a PermissionError
What if the file doesn't exist? This could throw FileNotFoundError

This can be handled in a try/except block

try: 
    -- do my things for file 
except FileNotFoundError:
    Specifically handle what happens for the file not existing
except PermissionError:
     Okay, what to do for permission error

-- this allows you to handle both exceptions if each needed a different handle
or
except (PermissionError, FileNotFoundError):
   both can be handled the same way

Reply:
Try blocks are part of my next learning block that I want to use and understand. So thanks for bringing it up.